doc/conf.py

```diff
@@ -227,6 +228,10 @@
 tokenizer_lang = "en_UK"
 spelling_word_list_filename = ["spelling_wordlist.txt"]
 
+# sphinx_js options
+js_language = 'typescript'
+js_source_path = '../tests/npm-app/src/ccf'
```

Comment:
Not great, but also not *very bad*. Should be done differently in the future.

Reply:
Can we move the sources out of the tests directory? Maybe simply `src/js`?

Reply:
Once this is distributed as a library on npm it can move to a separate folder, but for now let's keep it there, otherwise the JS sample apps are not copy-pasteable.

Reply:
We could move it to src/js and have a symlink in the samples.

Reply:
Yeah, I like that too.

Reply:
OK, let's do that.. if you then  `cp -rf` a sample you will still have a symlink to src/js which then takes effort to un-symlink.